### PR TITLE
Feature/in viewport

### LINKED ISSUE #12: Check if element is partly or almost visible in viewport
Found myself wanting this in a project.

Examples:
- `isElementInViewport(el, 200)`: `true` when element `el` is at least 200px inside the viewport
- `isElementInViewport(el, -200)`: `true` when element `el` is at most 200px outside the viewport

### PR DESCRIPTION
**Pull request by asbjornh**
Fixes #12 

* Refactors `isElementInViewport` and renames it to `isFullyInViewport`
* Adds `isInViewport`

The introduction of an offset led to destructive changes in the API. The previous implementation did not make sense in combination with an offset. An `undefined` offset would require the element to be fully visible, while adding a very small offset would let it be just inside of the viewport. A huge change in behaviour for very small parameter values (and smaller changes for larger parameter values) feels counter intuitive.

Also, it turns out that if the implementation only checks whether an element is x or y pixels inside of the viewport, the fully-in-viewport test is just a special case where the x- and y-offsets are equal to the dimensions of the element!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,24 +127,35 @@ Returns the value of the first DOM node with `dataAttributeName`
...
+
+_@creuna/utils/is-equal_
+
+* `a`: any
+* `b`: any
+* returns: boolean
+
+Checks whether `a` and `b` are equal (deep comparison for objects and arrays). This uses `JSON.stringify`, so be aware that array elements or object values that are `undefined` will be stripped.
+
+### isFullyInViewport(_node_, _offset_)
```
Comment (pellebjerkestrand) [inline README.md]: This function doesn't take an offset.

File: README.md [hunk 2]
```diff
@@ -127,24 +127,35 @@ Returns the value of the first DOM node with `dataAttributeName`
...
+_@creuna/utils/is-fully-in-viewport_
 
 * `node`: DOM node
 * returns: boolean
 
-Checks whether the given element is fully visible in the viewport
+Checks whether the given element is fully visible in the viewport. This is a special case of `isInViewport` where the offsets are the dimensions of the element.
 
-### isEqual(_a, b_)
+### isInViewport(_node_, _offset_)
```
Comment (pellebjerkestrand) [inline README.md]: Thus function also takes `offsetX`.

File: README.md (hunk 1 above)
Comment (asbjornh) [reply README.md]: l o l

File: README.md (hunk 2 above)
Comment (asbjornh) [reply README.md]: lål

File: README.md [hunk 3]
```diff
@@ -127,24 +127,35 @@ Returns the value of the first DOM node with `dataAttributeName`
...
+_@creuna/utils/is-fully-in-viewport_
 
 * `node`: DOM node
 * returns: boolean
 
-Checks whether the given element is fully visible in the viewport
+Checks whether the given element is fully visible in the viewport. This is a special case of `isInViewport` where the offsets are the dimensions of the element.
 
-### isEqual(_a, b_)
+### isInViewport(_node, offset, offsetX_)
```
Comment (pellebjerkestrand) [inline README.md]: `isInViewport(_node, offset, offsetX_)` should be `isInViewport(_node, offset[, offsetX]_)`, since `offsetX` is optional.

**Comment by pellebjerkestrand** [comment]
This now has merge conflicts that need to be resolved.

**Comment by asbjornh** [comment]
> This now has merge conflicts that need to be resolved.

Done! :)

**Comment by asbjornh** [comment]
Merging since all review comments have been resolved

**Pull request merged**